numpy/add_newdocs.py

```diff
@@ -5843,6 +5843,12 @@ def luf(lamdaexpr, *args, **kwargs):
         the result will broadcast correctly against the original `arr`.
 
         .. versionadded:: 1.7.0
+    initializer : scalar, optional
+        The initializer before starting the reduction. The reduction starts
```

Comment:
How about

> The value to which the result is initialized before starting the reduction. If not specified, the identity for this ufunc is used. For ufuncs with no identity like `maximum`, a reduction cannot be performed over an empty axis unless this argument is provided.

Reply:
> If None

Would "if not supplied" be better for object arrays in the future?

Reply:
My comment above describes what actually happens if `None` is passed, I think - it may not be what we want, but it's what we have right now.

I think the eventual fix here is to provide `np.ufunc.NoIdentity` as a sentinel that we use in place of the current `None`

Reply:
I think that should technically go to #8955.

Reply:
One final remark for this docstring
```
If arr.dtype is object, then the initializer is _only_ used when the reduction is empty
```

This is perhaps not ideal, but importantly it _is_ useful enough to fix `linalg.norm`

Reply:
I expected this to not work for empty `object` arrays at all, but it seems you're right! I've updated the docstring.

```python
>>> class A(object):
...     pass
... 
>>> np.sum(np.array([], dtype=object), initializer=A())
<A object at 0x1023abf28>
```

Reply:
I was expecting exactly the same thing, and was half-way through writing about how that was going to be really annoying before realizing we can sort-of get away with it